### DISCUSSION TITLE
ps_filter by group?

### DISCUSSION THREAD

**Discussion by antoine4ucsd** [david-barnett/microViz#173] [Q&A]
Hello
first, thank you for this impressive suite of tools!
I am trying to clean up my set of gut microbiome data. this includes multiple subjects and multiple tissues 

For filtering, I am considering 2 approaches - not sure which one is optimal

- 1. applying the filtering criteria on all
-  2. split the data by tissue and then filter 

For (1) 
```
# Prevalence threshold
prevalence_threshold <- 0.05  # 5% of samples
# Detection threshold for prevalence
detection_threshold <- 10  # 10 reads
# Total abundance threshold
total_abundance_threshold <- 100  # 100 total reads across all samples
# Sample abundance threshold
sample_abundance_threshold <- 10  # 10 reads in at least one sample
data|>
        tax_filter(min_prevalence = prevalence_threshold, 
                   prev_detection_threshold = detection_threshold, 
                   min_total_abundance = total_abundance_threshold, 
                   min_sample_abundance = sample_abundance_threshold)

```

is there a way to do it for (2) (group by?)

would you have any advice on the best approach? optimal filtering criteria? I assume (2) will likely be impacted by the prevalence_threshold ...

thank you

**Comment by david-barnett** [comment]
Hi Antoine, I only just spotted this message.
Glad you're finding microViz useful. I'll reply in case my thoughts are still useful for yourself or others.

I think there are maybe two rationales for filtering taxa by prevalence or abundance.

1. removing sequences /taxa that are likely erroneous

- hopefully most of this stuff is gone after using dada2 or similar - maybe also decontam
- but for this rationale i would recommend using a light prevalence filter across all samples

2. focusing a statistical analysis on relevant taxa

- in differential abundance analysis, you are often less likely to be interested in rare taxa
- moreover, test performance will be worse on rare taxa e.g. prev < 10%
- in this case, I might filter per group, if you will then test within that group
- if you are going to look for differences between those tissues/groups then definitely DO NOT filter them separately, as that would obviously cause artificial differences to appear
